src/Persistent/Idiom.hs

```diff
@@ -127,3 +127,10 @@ setTopicPhase tid phase = modifyTopic tid $ topicPhase .~ phase
 deactivateUser :: AUID User -> AUpdate ()
 deactivateUser uid
     = modifyUser uid (userSettings . userSettingsPassword .~ UserPassDeactivated)
+
+getUserViews :: Query [UserView]
+getUserViews = fmap2 makeUserView getAllUsers
+
+findActiveUser :: AUID User -> EQuery User
+findActiveUser =
+    (maybe404 . (^? activeUser) . makeUserView) <=< maybe404 <=< findUser
```

Comment:
I would like to separate cases, but we don't have different exception for each cases, yet.